### ISSUE TITLE
When editing a song, cache is not rebuilt

### ISSUE THREAD

**Issue by bloodybowlers** [bloodybowlers/UkeGeeks-ng#45]
When editing a song, cache is not rebuilt.

For example, if I change the title of a song, it is not shown on the song list, unless I refresh the cache (by calling "reindex").

**Comment by bloodybowlers** [comment]
I guess it was done for performance reason (not rebuilding the WHOLE cache for just one song change)  :(

**Comment by Louis-coding-UGNG** [comment]
The underlying question is: how often do you change the song header data (title, artist, album, maybe key, et al.)? I would assume most of the changes are done in the song body, correcting typos and probably changing some chords.
So **not** automatically rebuilding the whole cache on every edit is perhaps a good idea - I'd even say you could consider this a desired behaviour with no need to change.

Giving users with edit rights a "rebuild cache" button should really be sufficient.

**Comment by bloodybowlers** [comment]
> So not automatically rebuilding the whole cache on every edit is perhaps a good idea - I'd even say you could consider this a desired behaviour with no need to change.

Yes. But one thing that bother me is : the user doesn't know that if it changes the title (or artist, ...) of the song, it won't get updated on the song list page if he doesn't rebuild the cache.
We should be able to detect those changes  (when hitting the SAVE button) and either, rebuilding the cache, or telling the user to do it.
